File: ios/Storehop/UI/Util/EmptyState.swift
import SwiftUI

/// Shared empty-state surface for the Items, Shop-at-Store, and Store
/// Picker screens (and any future "nothing here yet" surface). Mirrors the
/// Android `EmptyState` Composable from `ui/util/EmptyState.kt` so the two
/// ports look consistent.
///
/// Layout: full-width VStack with a large SF Symbol (~64pt, tinted
/// `onSurfaceVariant`), a `titleMedium` title, and a `bodyMedium` body
/// (both centered). Drop it inside a `Section { ... }` when used in a
/// SwiftUI `List`, or as a standalone view otherwise.
///
/// Caller supplies the SF Symbol name + two already-resolved strings.
/// The second string is called `message` (not `body`) because `body`
/// is reserved by the `View` protocol — Android calls it `body`.
struct EmptyState: View {
    let systemImage: String
    let title: String
    let message: String

    var body: some View {
        VStack(spacing: 12) {
            Image(systemName: systemImage)
                .font(.system(size: 64, weight: .light))
                .foregroundStyle(StorehopColors.onSurfaceVariant)
            Text(title)
                .font(StorehopTypography.titleMedium)
                .foregroundStyle(StorehopColors.onSurface)
                .multilineTextAlignment(.center)
            Text(message)
                .font(StorehopTypography.bodyMedium)
                .foregroundStyle(StorehopColors.onSurfaceVariant)
                .multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 32)
        .padding(.horizontal, 24)
    }
}

File: app/src/main/java/com/storehop/app/ui/util/EmptyState.kt
package com.storehop.app.ui.util

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

/**
 * Shared empty-state surface for the Items tab, the Shop tabs (StorePicker
 * + Shop-at-Store), and any future "nothing here yet" surface. Replaces the
 * inline text-only blocks that previously lived in each screen.
 *
 * Layout: full-screen Box with a centered Column containing a 96dp tinted
 * Material Symbols icon, a `titleMedium` title, and a `bodyMedium` body —
 * both colored `onSurfaceVariant` so they recede on light backgrounds and
 * stay legible on dark.
 *
 * Caller supplies the icon + the two strings (already-resolved, not res
 * ids) so the same Composable handles "no items yet" and "no matches for
 * X" with different visuals.
 */
@Composable
fun EmptyState(
    icon: ImageVector,
    title: String,
    body: String,
    modifier: Modifier = Modifier,
) {
    Box(
        modifier = modifier.fillMaxSize().padding(24.dp),
        contentAlignment = Alignment.Center,
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onSurfaceVariant,
                modifier = Modifier.size(96.dp),
            )
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface,
                textAlign = TextAlign.Center,
            )
            Text(
                text = body,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
            )
        }
    }
}
